logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/GetNotificationsUseCase.kt

```diff
@@ -0,0 +1,92 @@
+package com.wire.kalium.logic.feature.message
+
+import com.wire.kalium.logic.data.conversation.Conversation
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.message.MessageContent
+import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.logic.data.notification.DbNotificationConversation
+import com.wire.kalium.logic.data.notification.DbNotificationMessage
+import com.wire.kalium.logic.data.notification.DbNotificationMessageAuthor
+import com.wire.kalium.logic.data.publicuser.model.OtherUser
+import com.wire.kalium.logic.data.user.UserRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flatMapMerge
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.take
+
+class GetNotificationsUseCase(
```

Comment:
I think we could extract this to an interface and mark this implementation as internal, so we can mock it and test it later on android-reloaded